### PR TITLE
fix create table and executeQuery broadcast bug

### PR DESCRIPTION
**Pull request by zzbdatatom**
PR checklist:
- Did you run ClangFormat ? yes
- Did you run CheckStyle ? yes
- Did you check the comment / log / exception conventions in Engineering code process wiki page ? yes 
- Did you import unnecessary headers ? no 
- Did you surround `Daisy : starts/ends` for new code in existing ClickHouse code base ? yes

Please write user-readable short description of the changes:
1. fix dafault value is empty when create or update table
2. fix the judgment logic of executeQuery broadcast

### CODE REVIEW

File: src/Server/RestRouterHandlers/ColumnDefinition.cpp [hunk 1]
```diff
@@ -21,7 +21,11 @@ String getCreateColumnDefination(const Poco::JSON::Object::Ptr & column)
 
     if (column->has("default"))
     {
-        column_definition.push_back(" DEFAULT " + column->get("default").toString());
+        String default_str = "''";
+        if (!column->get("default").toString().empty())
+            default_str = column->get("default").toString();
```
Comment (hzhzhang) [inline src/Server/RestRouterHandlers/ColumnDefinition.cpp:26]: Confirm is it possible to use get("key", default_value) here?

Comment (zzbdatatom) [reply src/Server/RestRouterHandlers/ColumnDefinition.cpp:26]: No

File: src/Interpreters/executeQuery.cpp [hunk 2]
```diff
@@ -93,9 +93,25 @@ namespace
 /// broadcast the table definitions to notify all CatalogService
 void broadcastCatalogIfNecessary(const ASTPtr & ast, ContextPtr & context)
 {
-    if (!context->isDistributed() || context->isDistributedDDLOperation())
+    if (context->isDistributedDDLOperation())
```
Comment (chenziliang) [inline src/Interpreters/executeQuery.cpp:96]: For all distributed DDL, we shall return directly ? in which case, we will need broadcast the catalogs ? What's the problem with previous logic ?

Comment (zzbdatatom) [reply src/Interpreters/executeQuery.cpp:96]: 1. Only in the first distributed DDL, we will return directly.  the second DDL will carry the distributed_ddl flag
2. When the second ddl  and the table  is DistributedMergeTree engine, and any time it is not DistributedMergeTree, it needs to be broadcast
3. The previous logic will cause catalog cannot be synchronized after non-DistributedMergeTree table been deleted through rest api, because the table is considered the first ddl

Comment (chenziliang) [reply src/Interpreters/executeQuery.cpp:96]: if we are not in distributed env (no Kafka is deployed), we shall return directly ? `context->isDistributed()` is generally for this unless is temporary disabled. 

We actually don't quite care if the storage engine is DistributedMergeTree or plain MergeTree.

So for plain MergeTree, when it gets dropped via REST API (our REST API ?), what the process will be ?

Our rest api accepts the request, since it is MergeTree, do we send this request to dwal and then DDLService picks it up and then send the drop request to the target box as DistributedMergeTree does ?

Comment (zzbdatatom) [reply src/Interpreters/executeQuery.cpp:96]: 1. First of all, in the Interpreter processing layer, if it is (1). isDistributed (2). isDistributedDDLOperation (3). engine == "DistributedMergeTree"， when the three conditions are met at the same time, it will generate the first stage of DDL with Fakfa, and put the DDL information in Kafka for persistence Transform
2. The second stage is that ddlservice consumes data from the Kafka __system_ddls topic and performs distributed processing. The sent request carries distributed_ddl = true to inform each node that this processing is the second stage

3. The current ddl processing process is that all ddl operations through REST API, if there is no parameter distributed_ddl = true in the request, it will be considered that the first stage of the interaction between Interpreter and kafka will be context->setDistributedDDLOperation (true), to avoid doBroadcast is unnecessary this time.

4. Drop mergree table Whether you are in distributed env or not, we all need to doBroadcast to modify the catalog at this time.
5. When using REST API, because there is no distributed_ddl = true parameter, context->setDistributedDDLOperation(true) will be set, so that you can never enter the doBroadcast

So here we need to separate DistributedMergeTree from plain MergeTree, and it doesn’t matter whether it’s distributed or not. 
Simply put, this bug is caused by the fact that the distributed_ddl = true parameter is not carried when operating the plain MergeTree through REST API.

Comment (chenziliang) [reply src/Interpreters/executeQuery.cpp:96]: If it `isDistributed` is true, can we do the exact same CRUD process as DistributedMergeTree for MergeTree ?

Comment (zzbdatatom) [reply src/Interpreters/executeQuery.cpp:96]: No we can't，We only support `DistributedMergeTree` table engine  in DDL Interpreter  for now. We can leave a fixme here and wait next pr to exact same CRUD process

Comment (chenziliang) [reply src/Interpreters/executeQuery.cpp:96]: If we can't, we don't need fix this for now since in the future we will need revert this again ?

Comment (zzbdatatom) [reply src/Interpreters/executeQuery.cpp:96]: No, we need fix this for now,  we don’t need evert this again later
1. First of all, we need to confirm that we only supports DistributedMergeTree table in REST API and DDL service
2. Only the DistributedMergeTree table will interact with kafka，so this place to judge whether it belongs to the second stage and engine is DistributedMergeTree
3.  plain MergeTree table also need to be board when doing ddl operations

Comment (chenziliang) [reply src/Interpreters/executeQuery.cpp:96]: This may not be true. We probably need support MergeTree creation via DDL service

**Comment by zzbdatatom** [comment]
Delayed development related functions
